packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1227,16 +1262,41 @@ class ProjectFileInvalidator {
         }
       }
     }
+    // We need to check the .packages file too since it is not used in compilation.
```

Comment:
I haven't made it down to the tests yet, but just making a note that it seems like this would need a test.

Reply:
Done